rotatedSortedArray.java

```diff
@@ -0,0 +1,40 @@
+// Time Complexity : O(logN)
+// Space Complexity : O(1) constant
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : understanding logic inside first else condition. It could have been
+//   if(target > nums[low] && target <= nums[mid]){
```

Comment:
See, if target is at mid we are returning mid, but what if target > nums[low] || target is at nums[low] itself, in that case we are searching towards left right. or target is lesser than element at mid